packages/functional/src/pipe.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * General pipe function.
+ * Provide an initial value and a list of functions to pipe it through.
+ * @param init  The initial value
+ * @param fns   Any number of functions to pipe the value through
+ * @returns     The final value with all functions applied
+ */
+export function pipe<TInitial>(init: TInitial): TInitial;
+export function pipe<TInitial, R1>(init: TInitial, init_r1: (init: TInitial) => R1): R1;
+export function pipe<TInitial, R1, R2>(init: TInitial, init_r1: (init: TInitial) => R1, r1_r2: (r1: R1) => R2): R2;
+export function pipe<TInitial, R1, R2, R3>(
+    init: TInitial,
+    init_r1: (init: TInitial) => R1,
+    r1_r2: (r1: R1) => R2,
+    r2_r3: (r2: R2) => R3
+): R3;
+export function pipe<TInitial, R1, R2, R3, R4>(
+    init: TInitial,
+    init_r1: (init: TInitial) => R1,
+    r1_r2: (r1: R1) => R2,
+    r2_r3: (r2: R2) => R3,
+    r3_r4: (r3: R3) => R4
+): R4;
```

Comment:
People are generally going to need way more overloads than this, for any kind of serious transaction. I don't know how to think about where to stop.

Reply:
Ahh yeah I was going to ask. I stopped at 4 for the tests, and figured we add more if we need. But I guess client-facing we don't know that upper bound.

This is a tough position to be in, but worst case if someone needs more, they can always nest a pipe in the initial value.

How about arbitrarily starting with 10? 😅

Reply:
Maybe just figure out what approach other functional programming utility libraries took and follow suit?